host_core/lib/host_core/control_interface/server.ex

```diff
@@ -68,129 +75,138 @@ defmodule HostCore.ControlInterface.Server do
-            }
-
-          {:error, e} ->
-            %{
-              host_id: host_id,
-              provider_ref: start_provider_command["provider_ref"],
-              error: "Failed to start provider: #{e}"
-            }
-        end
+    start_provider_command = Jason.decode!(body)
 
-      {:reply, Jason.encode!(ack)}
-    else
-      {:reply,
-       Jason.encode!(%{
-         host_id: host_id,
-         failure: "Command received by incorrect host and could not be processed"
-       })}
-    end
+    ack =
+      case HostCore.Providers.ProviderSupervisor.start_provider_from_oci(
+             start_provider_command["provider_ref"],
+             start_provider_command["link_name"]
+           ) do
+        {:ok, _pid} ->
+          %{
+            host_id: host_id,
+            provider_ref: start_provider_command["provider_ref"]
+          }
+
+        {:error, e} ->
+          %{
+            host_id: host_id,
+            provider_ref: start_provider_command["provider_ref"],
+            error: "Failed to start provider: #{e}"
+          }
+      end
+
+    {:reply, Jason.encode!(ack)}
   end
 
   # Stop Provider
-  defp handle_request({"cmd", host_id, "sp"}, body, _reply_to) do
-    if host_id == HostCore.Host.host_key() do
-      stop_provider_command = Jason.decode!(body)
+  defp handle_request({"cmd", _host_id, "sp"}, body, _reply_to) do
+    stop_provider_command = Jason.decode!(body)
 
-      HostCore.Providers.ProviderSupervisor.terminate_provider(
-        stop_provider_command["provider_ref"],
-        stop_provider_command["link_name"]
-      )
+    HostCore.Providers.ProviderSupervisor.terminate_provider(
+      stop_provider_command["provider_ref"],
+      stop_provider_command["link_name"]
+    )
 
-      {:reply, Jason.encode!(%{})}
-    else
-      {:reply,
-       Jason.encode!(%{
-         host_id: host_id,
-         failure: "Command received by incorrect host and could not be processed"
-       })}
-    end
+    {:reply, Jason.encode!(%{})}
   end
 
   # Update Actor
   # input: %{"new_actor_ref" => "... oci URL ..."} , public key, etc needs to match a running actor
-  defp handle_request({"cmd", host_id, "upd"}, body, _reply_to) do
-    if host_id == HostCore.Host.host_key() do
-      update_actor_command = Jason.decode!(body)
+  defp handle_request({"cmd", _host_id, "upd"}, body, _reply_to) do
+    update_actor_command = Jason.decode!(body)
 
-      response =
-        case HostCore.Actors.ActorSupervisor.live_update(update_actor_command["new_actor_ref"]) do
-          :ok -> Jason.encode!(%{accepted: true})
-          :error -> Jason.encode!(%{accepted: false})
-        end
+    response =
+      case HostCore.Actors.ActorSupervisor.live_update(update_actor_command["new_actor_ref"]) do
+        :ok -> Jason.encode!(%{accepted: true})
+        :error -> Jason.encode!(%{accepted: false})
+      end
 
-      {:reply, response}
-    end
+    {:reply, response}
```

Comment:
re: above comment, it really seems like we vary from `""` to `%{}` to a full response object, I propose that all of our responses are the same JSON shape and just indicate success and provide an error message if applicable. Thoughts?